### ISSUE TITLE
Allow compilers to use multiple processors to build source

### ISSUE THREAD

**Issue by cugone** [lairworks/nas2d-core#136]
Linux:
Change the makefile command from `make -k` to `make -k --jobs`.

Windows:
Change the project property `Multi-Processor Compilation` to `Yes /MP`.
 1. Right-click on project in solution explorer.
 2. Select "Properties"
 3. Set "Configuration" drop-down to "All Configurations"
 4. Set "Platform" drop-down to "All Platforms"
 3. Drill down the property page tree on the left: `Configuration Properties > C/C++ > General`
 4. Change the setting `Multi-processor Compilation` to `Yes /MP`

This will make compiling considerably faster. Right now clean builds are only compiling one file at a time and it's starting to show a noticeable slowdown.

**Comment by DanRStevens** [comment]
This could be very useful to decrease build time on multiprocessor machines.

One concern here is limiting the number of concurrent tasks on more limited hardware. I think the CI machines might only allow 1 or 2 CPUs. If unlimited parallelism was enabled, it could try to compile all files in parallel, which might overload the machine. That could result in slower compiles, or compiler crashes if the system doesn't have enough memory to compile everything at the same time.

Using `make -j` without a number following the `j` will enabled unlimited parallelism. To limit parallelism, a number can be added following the `-j`. Example: `make -j4`

Side note: I should probably have used `--keep-going` rather than `-k` in the config file. That would be more self documenting. Using `-k` is more appropriate when you're manually typing something at the command line, and want to be quick about it. Similarly, using `--jobs` as you've done here is better than my habit of using `-j`.

I'm unaware if Visual Studio has any built in control to limit parallelism. I would appreciate someone testing this.

**Comment by cugone** [comment]
@DanRStevens 

The /MP switch defaults to "number of effective processors" unlike make's "number of recipes in the make file". So if you only have 6 cores, I have 16, and the CI has 2: running the build on your machine will use 6, mine will use 16 and the CI will use 2.

It can still be limited by providing a number from 1 to 65536 after the switch: `/MP4`

See Also:
https://docs.microsoft.com/en-us/cpp/build/reference/mp-build-with-multiple-processes?f1url=https%3A%2F%2Fmsdn.microsoft.com%2Fquery%2Fdev15.query%3FappId%3DDev15IDEF1%26l%3DEN-US%26k%3Dk(VC.Project.VCCLCompilerTool.MultiProcessorCompilation)%26rd%3Dtrue&view=vs-2019

**Comment by DanRStevens** [comment]
Ok, that's some good info. Let's push this forward and try this in a branch for Windows only. It would be nice to get some timing information from the CI builds to see how this affects performance. Be sure to tag this issue in the pull request, or in a commit message.

Later, we can do similar for Linux in a separate PR, also tagging this issue. I expect the Linux case may need more tweaking and verification, so let's keep it separate from the Windows changes.

**Comment by cugone** [comment]
I just ran multiprocessor compilation in AppVeyor on my own, albeit much larger, engine and double-checked the build environments in AppVeyor.

Compilation is faster but only by about half since the build environments only have 2-4 cores. (I think the 4 core machines are reserved for paying customers).

**Comment by DanRStevens** [comment]
That's a significant improvement. Will be good to get that in.

I suppose technically this is dependent on PR #140. We may need to wait until Brett gets that cleaned up and ready to merge.

**Comment by ldicker83** [comment]
I've set this flag in the Visual Studio 2019 project files (see PR #159). Waiting on Appveyor builds to complete to verify that everything builds properly.

When I merge in that PR I'll mark this issue closed.
